src/mono/mono/metadata/reflection.c

```diff
@@ -619,7 +619,7 @@ mono_method_get_object_handle (MonoMethod *method, MonoClass *refclass, MonoErro
 	if (!refclass)
 		refclass = method->klass;
 
-	return CHECK_OR_CONSTRUCT_HANDLE (MonoReflectionMethod, method, refclass, method_object_construct, NULL);
+	return CHECK_OR_CONSTRUCT_HANDLE (MonoReflectionMethod, m_method_get_mem_manager (method), method, refclass, method_object_construct, NULL);
```

Comment:
I wonder if this should look in the `refclass` memory manager - refclass can be a subclass of the declaring class, so  if it's in a different ALC, we want the memory manager lower down the class hierarchy, I think.

Same for the other MemberInfo reflection objects